frontend/src/app/item/item-cadastro/item-cadastro.component.ts

```diff
@@ -0,0 +1,159 @@
+  ];
+
+  constructor(
+    private itemService: ItemService, 
+    private fb: FormBuilder,
+    private notification: PageNotificationService
+    ) { }
+
+  iniciarForm() {
+    this.form = this.fb.group({
+      id: [null],
+      nome: ['', [Validators.required]],
+      descricao: ['', [Validators.required]],
+      foto: [null, [Validators.required]],
+      disponibilidade: [false, [Validators.required]],
+      situacao: ['OK', [Validators.required]],
+      idUsuario: [JSON.parse(localStorage.getItem("usuario")).id, [Validators.required]],
+      idCategoria: [null, {update: 'change'}]
+    });
+  }
+
+  ngOnInit(): void {
+    this.itemListagemOutput.emit();
+    this.iniciarForm();
+  }
+
+  showDialog(){
+    this.display = true;
+  }
+
+  onFileChanged(event){
+    const file = event.target.files[0];
+    file.name
+    const reader = new FileReader();
+    reader.onloadend = () => {
+      let image: string = `${reader.result}`;
+      this.form.patchValue({
+        foto: image.split(',',2)[1]
+      });
+      this.imagemUrl = file.name;
+    }
+    reader.readAsDataURL(file);
+  }
+
+  onSubmit() {
+    this.submit = true;
+    if(this.isEditing){
+      this.itemService.alterar(this.form.value).pipe(
+        finalize(() => {
+          this.submit = false;
+          this.fecharModal();
+        })
+      ).subscribe(
+        () => {
+          this.itemListagemOutput.emit();
+          this.notification.addSuccessMessage("Item Atualizado com Sucesso.");
+        },
+        () => {
+          this.notification.addErrorMessage("Falha ao Atualizar Item.");
+        }
+      );
+    }else{
+      this.itemService.salvar(this.form.value).pipe(
+        finalize(() => {
+          this.submit = false;
+          this.fecharModal();
+        })
+      ).subscribe(
+        (item) => {
+          this.itemListagemOutput.emit();
+          this.notification.addSuccessMessage("Item Inserido com Sucesso.");
+        },
+        () => {
+          this.notification.addErrorMessage("Falha ao Inserir Item.");
+        }
+      );
+    }
+  }
+
+  alterar(id: number){
+    this.isEditing = true;
+    this.itemService.obterPorId(id).subscribe(
+      (item: ItemModel) => {
+        this.display = true;
+        this.form.patchValue({
+          ...item
+        });
+      }
+    );
+  }
+
+  fecharModal() {
+    this.form.patchValue(
+      {id: null}
+    );
+    this.form.patchValue(
+      {nome: ''}
+    );
+    this.form.patchValue(
+      {descricao: ''}
+    );
+    this.form.patchValue(
+      {foto: null}
+    );
+    this.form.patchValue(
+      {disponibilidade: false}
+    );
+    this.form.patchValue(
+      {situacao: 'OK'}
+    );
+    this.form.patchValue(
+      {idUsuario: JSON.parse(localStorage.getItem("usuario")).id}
+    );
+    this.form.patchValue(
+      {idCategoria: null}
+    );
+    this.display = false;
+    this.isEditing = false;
+    this.myImageInputVariable.nativeElement.value = "";
```

Comment:
pra que serve isso? this.myImageInputVariable.nativeElement.value = ""; ?

Reply:
Associada à annotation @ViewChild, myImageInputVariable: ElementRef referencia "event.target.files". Utilizado para limpar os arquivos presentes, como correção ao conflito na inserção de novas imagens ao item.

Reply:
top